src/main/java/io/vertx/core/logging/helper/LogFormatter.java

```diff
@@ -0,0 +1,361 @@
+/*
+ * Copyright (c) 2009 Red Hat, Inc.
+ * -------------------------------------
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Apache License v2.0 which accompanies this distribution.
+ *
+ *     The Eclipse Public License is available at
+ *     http://www.eclipse.org/legal/epl-v10.html
+ *
+ *     The Apache License v2.0 is available at
+ *     http://www.opensource.org/licenses/apache2.0.php
+ *
+ * You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.core.logging.helper;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Inspired from LogBack and Slf4j<br>
+ * 
+ * @author Patrick Sauts
+ */
+final public class LogFormatter {
+  static final char PARAM_START = '{';
+  static final char PARAM_STOP = '}';
+  static final String PARAM_STR = "{}";
+  private static final char ESCAPE_CHAR = '\\';
+  private static final String SEPARATOR = ", ";
+
+  /**
+   * @param messagePattern
+   * @param arg
+   * @return
+   */
+  final public static LogTupple format(String messagePattern, Object arg) {
+    if(arg instanceof Object[]) return paramsFormat(messagePattern, (Object[]) arg);
+    return paramsFormat(messagePattern, new Object[] { arg });
+  }
+
+  /**
+   * @param messagePattern
+   * @param arg1
+   * @param arg2
+   * @return
+   */
+  final public static LogTupple format(final String messagePattern, Object arg1, Object arg2) {
+    return paramsFormat(messagePattern, new Object[] { arg1, arg2 });
+  }
+
+  /**
+   * @param messagePattern
+   * @param params
+   * @return
+   */
+  final public static LogTupple paramsFormat(final String messagePattern, final Object[] params) {
+
+    if (params == null) {
+      return new LogTupple(messagePattern);
+    }
+
+    Throwable throwableCandidate = getThrowableCandidate(params);
+    int length = params.length;
+    if(throwableCandidate != null) --length;
+    
+    if (messagePattern == null) {
+      return new LogTupple(join(params,length), throwableCandidate);
+    }
+
+
+    int i = 0;
+    int j;
+    StringBuilder sbuf = new StringBuilder(messagePattern.length() + 50);
```

Comment:
Well handling escapes and double escapes make things a little more tricky. And I actually took the same approach than the Slf4j team.
I actually have some code to do that with patterns:

```
private static final String PARAM_PREFIX = '{'
  private static final String PARAM_SUFFIX = '}'
  private static final String PARAM_REGEX = ???

  private def replaceParams(String template, List params){
    int index = 0;
    int paramIndex = 0;
    Matcher matcher = Pattern.compile(PARAM_REGEX).matcher(template);
    StringBuilder sb = new StringBuilder();
    while (matcher.find()) {
      int start = matcher.start();
      int end = matcher.end();
      sb.append(template.substring(index, start));
      index = end;
      String tagParam = template.substring(start, index);
      String value = params.size() > paramIndex?params.get(paramIndex):"{}";
      sb.append(value);
    }
    sb.append(template.substring(index))
    return sb.toString()
  }
```

The right PARAM_REGEX is to be  written though.